src/lxml/ElementInclude.py

```diff
@@ -202,13 +202,13 @@ def _include(elem, loader=None, base_url=None,
                 if max_depth == 0:
                     raise LimitedRecursiveIncludeError(
                         "maximum xinclude depth reached when including file %s" % href)
-                _parent_hrefs.add(href)
                 node = load_include(href, parse, parser=parser)
                 if node is None:
                     raise FatalIncludeError(
                         "cannot load %r as %r" % (href, parse)
                         )
-                node = _include(node, loader, href, max_depth - 1, _parent_hrefs)
+                _parent_hrefs_for_node = set([href]) | _parent_hrefs
```

Comment:
Removing the `href` from the set after the recursion seems a less involved operation than creating a new list and two sets on each include. (Not that it makes a real difference in terms of time or space, but still …)

Reply:
Yes, I thought of that. But I think with the current implementation it is easier to understand (and guaranteed), that the call of `_include` is not able to change the parent_hrefs of the caller. But if you disagree, I can change it like you proposed.

Reply:
I won't fight for it, but then please use at least a set literal.

Reply:
Good point! I did not know that this is faster.